pkg/config/output.go

```diff
@@ -162,12 +162,12 @@ func (p *PipelineConfig) updateEncodedOutputs(req EncodedOutput) error {
 		}
 	}
 
-	if segmentConf := p.Outputs[types.EgressTypeSegments]; segmentConf != nil && len(segmentConf) > 0 {
-		// double the segment length
-		p.KeyFrameInterval = float64(2 * segmentConf[0].(*SegmentConfig).SegmentDuration)
-	} else if p.KeyFrameInterval == 0 && p.Outputs[types.EgressTypeStream] != nil {
+	if p.KeyFrameInterval == 0 && p.Outputs[types.EgressTypeStream] != nil {
 		// default 4s for streams
 		p.KeyFrameInterval = 4
+	} else if segmentConf := p.Outputs[types.EgressTypeSegments]; segmentConf != nil && len(segmentConf) > 0 {
```

Comment:
We should make sure to test this change with the stream + segment and segments with a duration > 4s. In the past, I've seem issues where the splitmuxsink was generating segments of very uneven duration if the key frame interval was < the segment duration. 

Also note that splitmuxsink will force a key frame regardless of the p.KeyFrameInterval parameter when a segment split is needed.

Reply:
tested with 6s segment + stream, seems it was fixed in one of the gstreamer updates